packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -2017,14 +2019,16 @@ export class UUIDV4 extends AbstractDataType<string> {
   }
 }
 
+export type IncludeAsCallback = (includeAs: string) => [Literal | Fn, string];
```

Comment:
I also don't think it should receive the `includeAs` variable. It makes the logic much more complex, when instead most of the logic could be handled in `_injectDependentVirtualAttributes`

Being able to point to the correct attribute in a raw query is something that should be supported in all raw queries, not just `DataTypes.VIRTUAL`. I don't want to have a virtual-specific solution here as it will lead to conflicting solutions in the future.

We already worked on this with `sql.attribute`: https://648dd325deb93b14ad86d83a--sequelize-site.netlify.app/docs/v7/querying/raw-queries/#sqlattribute

Reply:
So it could be just an Expression. Doesn't need to be a callback and in the expression we use sql.attribute to get whatever we need, right?

Reply:
Oh yes good point, it doesn't even need to be a callback then